com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventProgress.cs

```diff
@@ -171,37 +221,40 @@ internal void SetSceneLoadOperation(AsyncOperation sceneLoadOperation)
         /// </summary>
         internal void SetSceneLoadOperation(ISceneManagerHandler.SceneEventAction sceneEventAction)
         {
-            sceneEventAction.Completed = SetComplete;
+            sceneEventAction.Completed = SetLoadCompleted;
```

Comment:
Ah, is this why SceneEventAction was changed to a class? Because the struct passed here was a copy and this was basically ending up NOP?

Would it be possible to take the struct by `ref` here and avoid making it a class?

Reply:
Good point on the ref...let me check that... and yes that was why I semi-punted since it is really only used during integration tests.

Reply:
After looking at this, the cost of the class malloc is tiny and occurs only during a scene load or unload (when there is typically much larger allocations).  
I could try to refactor this whole process but I am not sure the savings we will get during loading and unloading would be worth it.

Reply:
Did a refactor of that entire crazy process.
You get a 💯 for calling that out... forced me to further think about "what else is wrong with this picture".
👍